native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ArgumentHandler.java

```diff
@@ -91,7 +105,10 @@ private void populateArgumentsMap(BObject fieldNode) {
 
     private Object getArgumentValue(BObject argumentNode, Type parameterType) {
         if (isFileUpload(parameterType)) {
-            return getFileUploadParameter(argumentNode, parameterType);
+            return this.getFileUploadParameter(argumentNode, parameterType);
+        } else if (isRepresentationArgument(parameterType)) {
+            Object jsonRepresentation = this.getJsonArgument(argumentNode);
```

Comment:
Why do we need this as a `JSON`?

Reply:
representation argument is an array of JSON serializable object therefore argument node is parsed to json object first.
then converted to `Representation` record